### PR TITLE
Nj 106 eligibility screener

### PR DESCRIPTION
**Pull request by aloverso**
## Link to pivotal/JIRA issue
- https://github.com/newjersey/affordability-pm/issues/106

## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

## What was done?
- Add eligibility question for health insurance
- Add eligibility test including health insurance + exemption logic for offboarding

## How to test?
- `married_filing_jointly_spouse_claimed_dep` for no exemption
- `married_filing_jointly_primary_claimed_dep` for no exemption
- `married_filing_jointly_neither_claimed_dep` for no exemption
- `married_filing_jointly_both_claimed_dep` for exemption
- `claimed_as_dependent` for exemption, not MFJ
- Any other persona (ie, `minimal`) for no exemption, not MFJ

**NOTE** for each, you will have to edit the W2 wages to test the wage exemption. By default all WILL qualify for exemption due to low wages. Edit response XML and increase wages to test the claimed as dependent part of the exemption

### Eligible due to answering "Yes" to question:

https://github.com/user-attachments/assets/bbb99f16-d13f-4c1a-bd1d-74e06f0f5592

---

### Eligible due to income exemption

https://github.com/user-attachments/assets/edaf79b6-869b-4194-b07e-a3bf0ab45ec1

---

### Eligible due to both primary/spouse claimed as dependent exemption

https://github.com/user-attachments/assets/bb638450-3714-465e-8e84-4a137f5f1c52

---

### Ineligible

https://github.com/user-attachments/assets/92d73098-851c-4b1e-89ca-57c97593b620

### CODE REVIEW

File: app/views/state_file/questions/nj_eligibility_health_insurance/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            label_text: t(".label"),
+            collection: [
+              { value: "yes", label: t("general.affirmative") },
+              { value: "no", label: t("general.negative") },
+            ]
+          ) %>
+    </div>
+
+    <div class="reveal">
+      <p><button class="reveal__button"><%= t('.helper_heading') %></button></p>
```
Comment (mmazanec22) [inline app/views/state_file/questions/nj_eligibility_health_insurance/edit.html.erb:20]: [pebble] This button doesn't need to be in a p tag

File: config/locales/en.yml [hunk 2]
```diff
@@ -2370,6 +2370,7 @@ en:
             eligibility_withdrew_529: withdrew money from a 529 college savings account in 2023
             eligibility_withdrew_msa_fthb: withdrew money from a medical savings account or a first-time homebuyer account
             eligibility_yonkers: lived or earned income in Yonkers in 2023
+            health_insurance_eligibility: you and all members of your tax household did not have health insurance (with minimum essential coverage) for every month in 2024
```
Comment (mmazanec22) [inline config/locales/en.yml:2373]: [pebble] year can be a parameter instead

File: config/locales/en.yml [hunk 3]
```diff
@@ -2786,6 +2787,21 @@ en:
           primary_disabled_question: Are you disabled according to the New Jersey definition?
           spouse_disabled_question: Is your spouse disabled according to the New Jersey definition?
           title: Disabled Exemption
+      nj_eligibility_health_insurance:
+        edit:
+          label: Did you and all members of your tax household have health insurance (with minimum essential coverage) for every month in 2024?
```
Comment (mmazanec22) [inline config/locales/en.yml:2792]: Year could also be a parameter here

File: config/locales/en.yml [hunk 4]
```diff
@@ -2786,6 +2787,21 @@ en:
...
+          title: Let’s get started.
+          helper_description_html: |
+            <p>Most people who have health insurance have minimum essential coverage. “Minimal essential coverage” just refers to how much your plan covers. If you have any of the following, you most likely have it:</p>
+            <ul>
+              <li>A plan through your employer</li>
+              <li>A plan bought through <a href="https://nj.gov/getcoverednj/">GetCoveredNJ</a></li>
```
Comment (mmazanec22) [inline config/locales/en.yml:2798]: [pebble] these external links need rel="noopener noreferrer" and we may also want to do target="_blank" since being taken out of a workflow is pretty annoying

File: spec/controllers/state_file/questions/nj_tenant_eligibility_controller_spec.rb [hunk 5]
```diff
@@ -44,7 +44,7 @@
     end
 
     context "when a user is not MFS" do
-      let(:intake) { create :state_file_nj_intake, :df_data_mfj }
+      let(:intake) { create :state_file_nj_intake, :married_filing_jointly }
```
Comment (mmazanec22) [inline spec/controllers/state_file/questions/nj_tenant_eligibility_controller_spec.rb:47]: [dust] why this change?

Comment (aloverso) [reply spec/controllers/state_file/questions/nj_tenant_eligibility_controller_spec.rb:47]: I got rid of `df_data_mfj` altogether. To instead specifically have the DF traits indicate if spouse/primary is a claimed dependent. All these changes are referencing DF data when really all they care about is the filing status itself. No need to over-do the setup

File: app/controllers/state_file/questions/eligibility_offboarding_controller.rb [hunk 6]
```diff
@@ -5,12 +5,15 @@ class EligibilityOffboardingController < QuestionsController
...
+      before_action -> { @filing_year = Rails.configuration.statefile_current_tax_year }
+
       def ineligible_reason
         key = current_intake.disqualifying_eligibility_answer
         if key.present?
           I18n.t(
             "state_file.questions.eligibility_offboarding.edit.ineligible_reason.#{key}",
-            state: current_state_name
+            state: current_state_name,
+            filing_year: @filing_year
```
Comment (aloverso) [inline app/controllers/state_file/questions/eligibility_offboarding_controller.rb:16]: Want to flag this change for CFA, I noticed all the other eligibility offboarding reasons have years hardcoded in the `en.yml` and `es.yml` files.

Didn't want to presume that I should change all those as well to use `filing_year` instead, but wanted to flag in case you all did want to take that on.

Comment (mpidcock) [reply app/controllers/state_file/questions/eligibility_offboarding_controller.rb:16]: yes, that's being addressed, thank you for the flag!

File: spec/lib/submission_builder/ty2024/states/nj/nj_return_xml_spec.rb [hunk 7]
```diff
@@ -10,7 +10,7 @@
 
     describe "XML schema" do
       context "with JSON data" do
-        let(:intake) { create(:state_file_nj_intake, :df_data_mfj) }
+        let(:intake) { create(:state_file_nj_intake, :df_data_mfj_spouse_claimed_dep) }
```
Comment (mpidcock) [inline spec/lib/submission_builder/ty2024/states/nj/nj_return_xml_spec.rb:13]: [dust] is spouse_claimed_dep relevant for this test?

**Review by mpidcock — APPROVED** [review]
✅

**Comment by mluedke2** [comment]
Two notes on the videos in the PR description:

* The ones labelled `Eligible due to income exemption` and `Eligible due to both primary/spouse claimed as dependent exemption` seem to be the same video, and for the income exemption.

* Since the same XML is used in the one labelled `Eligible due to answering "Yes" to question` and `Eligible due to income exemption`, it's unclear whether the user is eligible because of their Yes answer because we know they will be eligible either way. Maybe the one highlighting the question's answer should use the same XML as the `Ineligible`?

File: app/models/state_file_nj_intake.rb [hunk 8]
```diff
@@ -150,10 +153,32 @@ def disqualifying_df_data_reason
...
+      has_exception ? "eligible" : "ineligible"
+    else
+      "eligible"
+    end
+  end
+
   def disqualifying_eligibility_rules
     {
-      eligibility_lived_in_state: "no",
-      eligibility_out_of_state_income: "yes",
```
Comment (mluedke2) [inline app/models/state_file_nj_intake.rb:156]: [dust, could be boulder] why were these two rules removed here?

Comment (mluedke2) [reply app/models/state_file_nj_intake.rb:161]: to bring this full circle -- i believe the explanation given in-person was that these two rules were not actually having an effect so could be safely removed. @aloverso please correct me if i'm misunderstanding
